src/stcal/resample/resample.py

```diff
@@ -1085,8 +1085,9 @@ def finalize_resample_variance(self, output_model):
             for varname in self.variance_array_names:
                 varwsum = self._variance_info[varname]['wsum']
                 weight = self._variance_info[varname]['wt']
-                output_model[varname] = (varwsum / (weight * weight)).astype(
-                    dtype=self.output_array_types[varname]
+                output_model[varname] = (varwsum / (weight * weight)
```

Comment:
I recommend not doing `x/y/z`.  Visually, this is ambiguous and could mean `x/(y/z)` or `(x/y)/z`.  It's the latter and mathematically equivalent to `x/(y*z)`.  I recommend changing the above `x/y/z` to `x/(y*z)` because it's easier to read and computationally a double divide is more expensive than one multiplication and one divide.

Reply:
Done as part of the refactor to handle imaging vs spectroscopic modes, see if you think this is more clear now.